### PR TITLE
try revert some some to make dm works

### PR DESCRIPTION
**Pull request by july2993**
revert some make dm deploy&start work, mainly ref: https://github.com/pingcap/tiup/pull/490

review recommends:
carefully review the little code changed not in /dm directory.
glance over the code about dm. (just uncomment some code and fix it int the quick way, may need more polish and refactor)

### CODE REVIEW

File: templates/scripts/run_dm-master.sh.tpl [hunk 1]
```diff
@@ -17,9 +17,9 @@ cd "${DEPLOY_DIR}" || exit 1
 {{- end}}
 
 {{- if .NumaNode}}
-exec numactl --cpunodebind={{.NumaNode}} --membind={{.NumaNode}} bin/dm-master \
+exec numactl --cpunodebind={{.NumaNode}} --membind={{.NumaNode}} bin/bin/dm-master \
```
Comment (lonng) [inline templates/scripts/run_dm-master.sh.tpl:20]: It's better to use `bin/dm-master` instead of `bin/bin/dm-master`

File: pkg/dm/spec/paths.go [hunk 2]
```diff
@@ -14,24 +14,8 @@
 package spec
 
 import (
-	"fmt"
+	"github.com/pingcap/tiup/pkg/meta"
```
Comment (lonng) [inline pkg/dm/spec/paths.go:17]: Prefer to delete this file.

File: pkg/dm/spec/topology_dm.go [hunk 3]
```diff
@@ -0,0 +1,742 @@
...
+		Worker map[string]interface{} `yaml:"dm-worker"`
+		Portal map[string]interface{} `yaml:"dm-portal"`
+	}
+
+	// DMTopologySpecification represents the specification of topology.yaml
+	DMTopologySpecification struct {
+		GlobalOptions GlobalOptions `yaml:"global,omitempty"`
+		// MonitoredOptions MonitoredOptions   `yaml:"monitored,omitempty"`
+		ServerConfigs DMServerConfigs    `yaml:"server_configs,omitempty"`
+		Masters       []MasterSpec       `yaml:"dm-master_servers"`
```
Comment (lonng) [inline pkg/dm/spec/topology_dm.go:92]: Mix the `_` and `-` is very weird.

File: pkg/dm/spec/topology_dm.go [hunk 4]
```diff
@@ -0,0 +1,742 @@
...
+	if err := fillDMCustomDefaults(&topo.GlobalOptions, topo); err != nil {
+		return err
+	}
+
+	return topo.Validate()
+}
+
+// platformConflictsDetect checks for conflicts in topology for different OS / Arch
+// for set to the same host / IP
+func (topo *DMTopologySpecification) platformConflictsDetect() error {
```
Comment (lonng) [inline pkg/dm/spec/topology_dm.go:285]: Can we reuse the `cluster/spec` conflicts check?

File: templates/scripts/run_dm-master.sh.tpl (hunk 1 above)
Comment (july2993) [reply templates/scripts/run_dm-master.sh.tpl:20]: https://github.com/pingcap/tiup/issues/557
add ` avoid use bin/bin/ in the deploy dir` in the issue, will not do this in this pr.
we always install the component in the {deploy_dir}/bin in the `InstallPackage` task.

File: pkg/dm/spec/topology_dm.go (hunk 4 above)
Comment (july2993) [reply pkg/dm/spec/topology_dm.go:285]: will do this in later pr, this file mainly copy from https://github.com/pingcap/tiup/pull/490/files
and fix many things not compile now...

File: pkg/dm/spec/paths.go (hunk 2 above)
Comment (july2993) [reply pkg/dm/spec/paths.go]: delete in https://github.com/pingcap/tiup/pull/554/commits/85e238f7b3cfc2daa91158971fa89eea9d3ed108

File: pkg/dm/spec/topology_dm.go (hunk 3 above)
Comment (july2993) [reply pkg/dm/spec/topology_dm.go]: https://github.com/pingcap/tiup/pull/554/commits/2788cbde31bcba797d9a98a99bff86f7bfd8cb89

**Comment by july2993** [comment]
PTAL @lonng I don't want to do refactor that will affect the current code about tidb-cluster at this pr.

**Pull request merged**